contracts/lz/LzApp.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@rari-capital/solmate/src/auth/Owned.sol";
+import "./interfaces/ILayerZeroReceiver.sol";
+import "./interfaces/ILayerZeroUserApplicationConfig.sol";
+import "./interfaces/ILayerZeroEndpoint.sol";
+
+/**
+ * @title LzApp
+ * @notice LayerZero-enabled contract that can have multiple remote chain ids.
+ * @dev this is a modified contract from the layerzero suggested implementation
+ */
+abstract contract LzApp is Owned, ILayerZeroReceiver, ILayerZeroUserApplicationConfig {
```

Comment:
Please, instead of creating a new file, let's import `LzApp` adding LayerZero repo as dependency